allennlp/training/trainer.py

```diff
@@ -178,8 +181,12 @@ def __init__(self,
             A ``Dataset`` to train on. The dataset should have already been indexed.
         validation_dataset : ``Dataset``, optional, (default = None).
             A ``Dataset`` to evaluate on. The dataset should have already been indexed.
-        patience : int, optional (default=2)
-            Number of epochs to be patient before early stopping.
+        patience : int != 0, optional (default=2)
+            If ``patience > 0``, it's the number of epochs to be patient before early stopping:
+            the training is stopped after ``patience`` epochs with no improvement.
+            If ``patience < 0 or patience == None``, early stopping is disabled.
```

Comment:
in `from_params` the default value is 2 if not specified, so how would patience be `None`?  I think that could only be the case if people explicitly put `"patience": null` in the config file? I don't know that I like the idea of having nulls in the config file